src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -920,20 +947,34 @@ update_estimate(TargetT &current_alpha_coefficent_image)
 
   base_type::compute_sub_gradient_without_penalty_plus_sensitivity (*multiplicative_update_image_ptr,
                                                           *current_update_image_ptr,
-                                                          subset_num); 
+                                                          subset_num);
+  unique_ptr< TargetT > kmultiplicative_update_ptr((*multiplicative_update_image_ptr).get_empty_copy());
+
+  const TargetT& sensitivity =
+    base_type::get_subset_sensitivity(subset_num);
 
+unique_ptr< TargetT > ksens_ptr(sensitivity.get_empty_copy());
+#ifdef STIR_OPENMP
+#pragma omp parallel sections
+ { omp_set_nested(1);
```

Comment:
no idea why you have `omp_set_nested`. It seems deprecated https://www.openmp.org/spec-html/5.0/openmpsu119.html and I guess replaced by `omp_set_max_active_levels`. However, https://www.openmp.org/spec-html/5.0/openmpsu125.html#x162-7300003.2.16 states

    This routine has the described effect only when called from a sequential part of the program. 
    When called from within a parallel or teams region, the effect of this routine is implementation defined. 

So this is probably ill-defined.

Reply:
In any case, this seems like a very bad idea. you're trying to do 2 kernelisations in parallel, but these are already parallelised. That might only be beneficial if you have many more threads then each "kernelisation" can use. But that won't be the case anymore with the collapsed-for loops. I'd remove the sections.